### PR TITLE
Introduce fixed collection form type

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets | -|
| License         | MIT |

Translations, channel configuration, promotion rules, promotion actions - all of them benefit by usage of fixed collection type.

### CODE REVIEW

File: src/Sylius/Bundle/ResourceBundle/Tests/Form/Type/FixedCollectionTypeTest.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    {
+        $form = $this->factory->create(FixedCollectionType::class, null, [
+            'entries' => ['first_name', 'last_name'],
+            'entry_type' => TextType::class,
+            'entry_name' => function ($entry) {
+                return strtoupper($entry);
+            },
+        ]);
+
+        $form->submit(['FIRST_NAME' => 'Elon', 'LAST_NAME' => 'Tusk']);
```
Comment (NeverResponse) [inline src/Sylius/Bundle/ResourceBundle/Tests/Form/Type/FixedCollectionTypeTest.php]: 🍾

Comment (Zales0123) [inline src/Sylius/Bundle/ResourceBundle/Tests/Form/Type/FixedCollectionTypeTest.php]: 🐃

Comment (pjedrzejewski) [inline src/Sylius/Bundle/ResourceBundle/Tests/Form/Type/FixedCollectionTypeTest.php]: XD

Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/Tests/Form/Type/FixedCollectionTypeTest.php]: 🐋

File: features/payment/managing_payment_methods/editing_payment_method.feature [hunk 2]
```diff
@@ -5,7 +5,8 @@ Feature: Editing payment methods
     I want to be able to edit payment method
 
     Background:
-        Given the store has a payment method "Offline" with a code "offline"
+        Given the store operates on a single channel in "United States"
```
Comment (pamil) [inline features/payment/managing_payment_methods/editing_payment_method.feature]: Don't know how this could work without any locale defined 🕴

**Comment by pjedrzejewski** [comment]
Thanks Kamil!

**Pull request merged**